config/openshift/v4_12_exp/translate.go

```diff
@@ -295,3 +296,24 @@ func validateMCOSupport(mc result.MachineConfig, ts translate.TranslationSet) re
 	}
 	return cutil.TranslateReportPaths(r, ts)
 }
+
+// fcos v1.5.0-exp generates a user.cfg file using append. Change this to contents
+func handleUserGrubCfg(config *types.Config, ts *translate.TranslationSet) {
+	for i, file := range config.Storage.Files {
+		if file.Path == "/boot/grub2/user.cfg" {
+			config.Storage.Files[i] = types.File{
+				Node: types.Node{
+					Path: "/boot/grub2/user.cfg",
+				},
+				FileEmbedded1: types.FileEmbedded1{
+					Contents: types.Resource{
+						Source:      file.Append[0].Source,
+						Compression: file.Append[0].Compression,
+					},
+				},
+			}
```

Comment:
We're still assuming that the fields we're ignoring are, in fact, empty.  The user might have put something in them.

Rather than checking the other fields one at a time, it might be cleaner to assign the entirety of `Append[0]` to `Contents` and update the TranslationSet accordingly.  We don't have a helper to do the latter right now, but `AddFromCommonSource` is close.  We could add a variant of `AddFromCommonSource` that appends subpaths to both the `From` and `To` side, instead of just the `To` side.